src/content/profile-data.js

```diff
@@ -131,9 +131,24 @@ export function defaultThreadOrder(threads: Thread[]) {
   return threadOrder;
 }
 
-export function filterThreadToJSOnly(thread: Thread) {
-  return timeCode('filterThreadToJSOnly', () => {
-    const { stackTable, funcTable, frameTable, samples } = thread;
+export function filterThreadByImplementation(thread: Thread, implementation: string): Thread {
+  const { funcTable } = thread;
+
+  switch (implementation) {
+    case 'cpp':
+      return _filterThreadByFunc(thread, funcIndex => !funcTable.isJS[funcIndex]);
+    case 'js':
+      return _filterThreadByFunc(thread, funcIndex => funcTable.isJS[funcIndex]);
+  }
```

Comment:
I'll probably leave this to a follow-up, when we want this filter to be smarter than it is.